packages/cli/templates/js/src/main.js

```diff
@@ -1,19 +1,192 @@
 import { createFieldPlugin } from '@storyblok/field-plugin'
 import './style.css'
 
-// This button is the only element in this application
-const buttonEl = document.createElement('button')
-buttonEl.classList.add('sb-button--primary')
-document.body.appendChild(buttonEl)
+let rootElement = document.querySelector('#app')
+if (!rootElement) {
+  rootElement = document.createElement('div')
+  rootElement.id = 'app'
+  document.body.appendChild(rootElement)
+}
+
+rootElement.innerHTML = `<span>Loading...</span>`
+let previousType = 'loading'
+let currentData
+
+const modalCloseButton = ModalCloseButton()
+const counter = Counter()
+const modalToggle = ModalToggle()
+const assetSelector = AssetSelector()
+const components = [modalCloseButton, counter, modalToggle, assetSelector]
 
 // Establish communication with the Visual Editor
 createFieldPlugin((response) => {
   // Re-render the button element when messages
-  const { data, actions } = response
-  buttonEl.textContent = `Increment: ${data.value ?? 0}`
-  buttonEl.onclick = () => actions.setValue((data.value ?? 0) + 1)
+  const { data, actions, type } = response
+  if (previousType === 'loading') {
+    previousType = type
+    if (type === 'error') {
+      rootElement.innerHTML = `<span>Error</span>`
+    } else if (type === 'loaded') {
+      renderFieldPlugin({ data, actions })
+    }
+  } else {
+    updateData(data)
+  }
 })
```

Comment:
Would it be possible to emphasise this part of the code by moving the rest of the logic to a separate file?